utils/logger.go

```
@@ -17,12 +20,36 @@ const (
 
 // NewLogger creates and returns a new instance of the Logger type. This
 // initializes the logger with a tagged logrus Entry initialized with the
-// version string and hostname.
-func NewLogger(version string) Logger {
+// version string, hostname and two logging options (syslog & logpath)
+// If syslog is true the logger will use local syslog method
+// If syslog is false logpath will be evaluated as log file with rotation
+// STDOUT is the logging fallback method
+func NewLogger(version string, syslogEnabled bool, logpath string) Logger {
 	log := logrus.New()
+	if syslogEnabled {
+		hook, err := logrus_syslog.NewSyslogHook("", "", syslog.LOG_INFO, "device-hub")
+		if err == nil {
```

Yeah, that has a lot of sense. The CLI should return the error to the caller and exit.